repose-aggregator/components/filters/openapi-validator-filter/src/main/scala/org/openrepose/filters/openapivalidator/HttpServletOAIRequest.scala

```diff
@@ -0,0 +1,150 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_=_
+ */
+package org.openrepose.filters.openapivalidator
+
+import java.io.InputStream
+import java.nio.charset.{IllegalCharsetNameException, StandardCharsets, UnsupportedCharsetException}
+import java.util
+import java.util.{Collections, Optional}
+
+import com.atlassian.oai.validator.model.Request
+import com.typesafe.scalalogging.slf4j.StrictLogging
+import javax.servlet.http.HttpServletRequest
+import org.openrepose.commons.utils.http.normal.{QueryParameter, QueryParameterCollection}
+import org.openrepose.filters.openapivalidator.HttpServletOAIRequest._
+
+import scala.collection.JavaConverters._
+import scala.io.Source
+
+/**
+  * A wrapper that exposes an underlying [[HttpServletRequest]] as a [[Request]]
+  * usable by an [[com.atlassian.oai.validator.OpenApiInteractionValidator]].
+  */
+class HttpServletOAIRequest(httpServletRequest: HttpServletRequest)
+  extends Request with StrictLogging {
+
+  private lazy val encoding: String = {
+    Option(httpServletRequest.getCharacterEncoding)
+      .getOrElse(StandardCharsets.ISO_8859_1.name)
+  }
+  private lazy val body: String = {
+    inputStreamToString(httpServletRequest.getInputStream, encoding)
+  }
+
+  // In accordance with OpenAPI specification, only query string parameters (not form data parameters)
+  // are made available:
+  // https://github.com/OAI/OpenAPI-Specification/blob/3.0.2/versions/3.0.2.md#parameter-locations
+  // For an example on validating form data parameters, see:
+  // https://github.com/OAI/OpenAPI-Specification/blob/3.0.2/versions/3.0.2.md#operation-object-example
+  private lazy val queryParameters: Seq[QueryParameter] = {
+    Option(httpServletRequest.getQueryString)
+      .map(new QueryParameterCollection(_))
+      .map(_.getParameters.asScala)
+      .getOrElse(Seq.empty)
+  }
+
+  // Header values are not split on commas.
+  // Splitting is an option when using the `SimpleRequest.Builder`, however doing so many not preserve the semantics
+  // of the split value.
+  // On the other hand, if we do not split, the validator may be unable to recognize multiple values and may
+  // treat them as a single value instead.
+  // For the moment, if splitting is necessary, the `SplitHeaderFilter` may be used.
+  // Additionally, header name lookups should be case-insensitive.
+  // This is achieved by simply lower-casing header names since there is no requirement to preserve casing.
+  private lazy val headers: Map[String, Seq[String]] = {
+    httpServletRequest.getHeaderNames.asScala.map(_.toLowerCase).map { headerName =>
```

Comment:
I actually tried to avoid wrapping the request unnecessarily. In fact, I don't necessarily know the request is an instance of our wrapper at any point in this filter's processing. I don't strictly need to work with our wrapper, and the `asScala` wrapper here is certainly lighter weight. I think I still prefer using `asScala` over the wrapper.

Reply:
yeah, i saw that further down that you only conditionally wrap it